database/src/lib.rs

```diff
@@ -13,20 +13,39 @@ extern crate regex;
 extern crate resopt;
 extern crate rocket;
 
+use std::env;
+use std::ops::Deref;
+
 pub use diesel::connection::Connection as DieselConnection;
+
+#[cfg(feature = "sqlite")]
+use diesel::sqlite::SqliteConnection;
+
+#[cfg(feature = "postgres")]
 use diesel::pg::PgConnection;
 use diesel::r2d2::{self, ConnectionManager};
 use rocket::http::Status;
+
 use rocket::request::{self, FromRequest};
 use rocket::{Outcome, Request, State};
-use std::env;
-use std::ops::Deref;
 
+use chrono_humanize::Humanize;
+
+pub mod datetime;
 pub mod idgen;
 pub mod models;
 pub mod schema;
 pub mod validators;
 
+#[cfg(all(feature = "sqlite", feature = "postgres"))]
+compile_error!("sqlite and postgres features cannot be simultaneously selected");
```

Comment:
i wish there was an easier (and prettier) way to do this… after all diesel doesn't use feature flags and can have all of its databases enabled at the same time.

Reply:
Diesel (or at least the version of Diesel we're using) has both `postgres` and `sqlite` feature flags.  The `postgres` and `sqlite` feature flags on the database crate toggle feature flags on `diesel` and `diesel_infer_schema`; we use them here as well.

I think there is a case to build in support for multiple servers for e.g. pre-built Rustodon binary distributions, but that seems to be some way off.  In the meantime, switching using types seemed like the mechanism that required the fewest code changes.

Reply:
i opened an issue for this in diesel https://github.com/diesel-rs/diesel/issues/1853